src/app/SimpleSubscriptionResumptionStorage.cpp

```diff
@@ -81,173 +81,229 @@ CHIP_ERROR SimpleSubscriptionResumptionStorage::SaveIndex(const SubscriptionInde
     return CHIP_NO_ERROR;
 }
 
+#define DeleteIndexAndReturnLogErrorOnFailure(expr)                                                                                                 \
+    do                                                                                                                             \
+    {                                                                                                                              \
+        auto __err = (expr);                                                                                                       \
+        if (!::chip::ChipError::IsSuccess(__err))                                                                                  \
+        {                                                                                                                          \
+mStorage->SyncDeleteKeyValue(DefaultStorageKeyAllocator::SubscriptionResumptionIndex().KeyName()); \
+ChipLogError(DataManagement, "%s at %s:%d", ErrorStr(__err), __FILE__, __LINE__);                                        \
```

Comment:
You cannot do this on embedded, it will start carrying __FILE__ which is a flash cost for nothing. Suggest just using CHIP_ERROR_FORMAT and err.Format().